src/lib/updating-element.ts

```diff
@@ -289,7 +296,8 @@ export abstract class UpdatingElement extends HTMLElement {
    */
   static createProperty(
       name: PropertyKey,
-      options: PropertyDeclaration = defaultPropertyDeclaration) {
+      options: PropertyDeclaration = defaultPropertyDeclaration,
+      descriptorFactory?: PropertyDescriptorFactory) {
```

Comment:
Rather than a function-valued argument, why not another overridable method? I'd rather be consistent in allowing customization via overrides than a mix of overrides and values. Unless there's a reason the factory needs to be effectively final.

Reply:
That's fair. The goal was to not change the shape of the existing API, which could be breaking, while providing the ability to customize. Currently `createProperty` does 4 things:

 1. stores property metadata
 1. decides whether or not to create an accessor
 1. creates a property descriptor 
 1. creates the accessor using the descriptor 

I considered making new entry points for (1) and (4) that overriders could simply call, without the need to call `super`; however, this meant users would have to do (2) on their own and always do (3) from scratch (cumbersome).

I also wanted to avoid reshaping the API, for example having `createProperty` return a descriptor, to avoid breaking users that are currently overriding `createProperty`. Admittedly it's currently difficult to override in a useful way, but it is possible.

Reply:
Removed factory in favor of `createPropertyDescriptor` which is called in the default implementation of `createProperty`.